apps/modernization-ui/src/apps/page-builder/page/management/edit/staticelement/EditStaticElement.tsx

```diff
@@ -0,0 +1,238 @@
+import { Button, Form, ModalRef, ModalToggleButton } from '@trussworks/react-uswds';
+import {
+    PageStaticControllerService,
+    PagesQuestion,
+    UpdateDefault,
+    UpdateHyperlink,
+    UpdateReadOnlyComments
+} from 'apps/page-builder/generated';
+import { SelectInput } from 'components/FormInputs/SelectInput';
+import { RefObject } from 'react';
+import { Controller, FormProvider, useForm } from 'react-hook-form';
+import { Input } from 'components/FormInputs/Input';
+import styles from './staticelement.module.scss';
+import { maxLengthRule } from 'validation/entry';
+import { CommentsFields } from './CommentsFields';
+import { HyperlinkFields } from './HyperlinkFields';
+import { useAlert } from 'alert/useAlert';
+import { usePageManagement } from '../../usePageManagement';
+import { authorization } from 'authorization';
+
+const staticType = [
+    { value: 'LIN', name: 'Line separator' },
+    { value: 'HYP', name: 'Hyperlink' },
+    { value: 'COM', name: 'Comments (read-only)' },
+    { value: 'PAR', name: 'Participant list (read-only)' },
+    { value: 'ELE', name: 'Electronic document list (read-only)' }
+];
+
+type EditStaticProps = {
+    modalRef?: RefObject<ModalRef>;
```

Comment:
Suggestion: It is typically better to not pass down modal refs. Perhaps the `onChange` can be leveraged to have the parent trigger the modal close as well as the `fetch` (or maybe even it just calls an `onChange`). That would make this component "dumber" so that it is only responsible for handling the `edit` and not the side affects.